src/ImageSharp/Processing/Processors/Binarization/BinaryThresholdProcessor{TPixel}.cs

```diff
@@ -90,14 +97,55 @@ public void Invoke(int y)
                 Span<TPixel> row = this.source.GetPixelRowSpan(y);
                 ref TPixel rowRef = ref MemoryMarshal.GetReference(row);
 
-                for (int x = this.minX; x < this.maxX; x++)
+                if (this.colorComponent == BinaryThresholdColorComponent.Luminance)
                 {
-                    ref TPixel color = ref Unsafe.Add(ref rowRef, x);
-                    color.ToRgba32(ref rgba);
+                    for (int x = this.minX; x < this.maxX; x++)
+                    {
+                        ref TPixel color = ref Unsafe.Add(ref rowRef, x);
+                        color.ToRgba32(ref rgba);
 
-                    // Convert to grayscale using ITU-R Recommendation BT.709 if required
-                    byte luminance = this.isAlphaOnly ? rgba.A : ColorNumerics.Get8BitBT709Luminance(rgba.R, rgba.G, rgba.B);
-                    color = luminance >= this.threshold ? this.upper : this.lower;
+                        // Convert to grayscale using ITU-R Recommendation BT.709 if required
+                        byte luminance = this.isAlphaOnly ? rgba.A : ColorNumerics.Get8BitBT709Luminance(rgba.R, rgba.G, rgba.B);
+                        color = luminance >= this.threshold ? this.upper : this.lower;
+                    }
+                }
+                else if (this.colorComponent == BinaryThresholdColorComponent.Saturation)
+                {
+                    float fThreshold = this.threshold / 255F;
+
+                    for (int x = this.minX; x < this.maxX; x++)
+                    {
+                        ref TPixel color = ref Unsafe.Add(ref rowRef, x);
+                        color.ToRgba32(ref rgba);
+
+                        // Extract saturation and compare to threshold.
+                        float sat = this.colorSpaceConverter.ToHsl(rgba).S;
+                        color = (sat >= fThreshold) ? this.upper : this.lower;
+                    }
+                }
+                else if (this.colorComponent == BinaryThresholdColorComponent.MaxChroma)
+                {
+                    float fThreshold = this.threshold / 2F;
```

Comment:
Hey @Franz1960 Can you explain this bit to me please? I'm struggling to figure out why we're halving the threshold.

Reply:
The achromatic level is in the center of the Cb and Cr range, so the distance is half the range.

Reply:
Thanks... Makes sense.

One last thing. Can you please give me push permissions to your fork? I just did some optimization work on the process but do not have permissions to push for some reason (_Normally I do with others_).

https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests

Reply:
The box had been checked:
![image](https://user-images.githubusercontent.com/19282511/108182490-59319c80-7109-11eb-8c4a-89cb49397df6.png)
but as it was not giving you the necessary rights (for whatever reason) I gave you write permission to the forked repo.